### PR TITLE
App shortcuts sample

### PR DESCRIPTION
**Pull request by gonzalonm**
Sample ported from https://github.com/googlesamples/android-AppShortcuts

### CODE REVIEW

File: android-n/AppShortcuts/AppShortcuts/Main.cs [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+		{
+			private Context Context { get; set; }
+			private LayoutInflater Inflater;
+			private List<ShortcutInfo> mList = EMPTY_LIST;
+			private Main Owner { get; set; }
+
+			public MyAdapter(Context context)
+			{
+				Context = context;
+				Inflater = (LayoutInflater)Context.GetSystemService(Java.Lang.Class.FromType(typeof(LayoutInflater)));
```
Comment (pjcollins) [inline android-n/AppShortcuts/AppShortcuts/Main.cs]: Can be `GetSystemService(Context.LayoutInflaterService)`

File: android-n/AppShortcuts/AppShortcuts/Main.cs [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+			editUri.Hint = "http://www.xamarin.com/";
+			editUri.InputType = InputTypes.TextVariationUri;
+
+			new AlertDialog.Builder(this)
+				.SetTitle("Add new website")
+				.SetMessage("Type URL of a website")
+				.SetView(editUri)
+				.SetPositiveButton("Add", (dialog, whichButton) =>
+				{
+					var url = editUri.Text.ToString().Trim();
```
Comment (pjcollins) [inline android-n/AppShortcuts/AppShortcuts/Main.cs]: Don't need a ToString () call on the Text property.

File: android-n/AppShortcuts/AppShortcuts/ShortcutHelper.cs [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+		private static long REFRESH_INTERVAL_MS = 60 * 60 * 1000;
+
+		private Context mContext;
+
+		private ShortcutManager mShortcutManager;
+
+		public ShortcutHelper(Context context)
+		{
+			mContext = context;
+			mShortcutManager = (ShortcutManager)mContext.GetSystemService(Java.Lang.Class.FromType(typeof(ShortcutManager)));
```
Comment (pjcollins) [inline android-n/AppShortcuts/AppShortcuts/ShortcutHelper.cs]: Can use `GetSystemService (Context.ShortcutService)` here.

**Comment by pjcollins** [comment]
Hey I noticed one issue with the port - the application crashes when you try to add too many app shortcuts (more then 5 I think?). This may just be a case of not catching the right exception type? The java version displays a toast with exception information in this case.
Stacktrace:
https://gist.github.com/pjcollins/f9358d1178235ed2a4e9d028dd8c6472

Other than that, there's a few minor stylistic changes
* We should remove the private keyword in variable declarations / method signatures which are private by default.
*  I've provided a couple inline comments for minor readability changes.

File: android-n/AppShortcuts/AppShortcuts/Main.cs (hunk 1 above)
Comment (gonzalonm) [inline android-n/AppShortcuts/AppShortcuts/Main.cs]: Done

File: android-n/AppShortcuts/AppShortcuts/Main.cs (hunk 2 above)
Comment (gonzalonm) [inline android-n/AppShortcuts/AppShortcuts/Main.cs]: Done

**Comment by gonzalonm** [comment]
Regarding with the app crashes when you try to add too many shortcuts, I've added a toast with the exception information.

Also, I've removed the private keyboard from variable declarations and method signatures.

**Comment by pjcollins** [comment]
WRT the try/catch for the Shortcut Manager invocation, I took a closer look at how the java lambdas were being used in the upstream sample and we can better emulate that to handle this scenario, please see the diff attached below:

https://gist.github.com/pjcollins/179c1e67a7d8e39189349408976f044b

**Comment by gonzalonm** [comment]
Changes applied

**Comment by pjcollins** [comment]
Looks good!

**Pull request merged**